OpenLaTeXStudio/EditorModule/src/latexstudio/editor/util/PropertyService.java

```diff
@@ -16,15 +18,45 @@
  */
 public final class PropertyService {
     
+    public static final String PROPERTIES_FILENAME = "latexProperties.txt";
+    
+    public static final String PROPERTIES_FILEPATH = System.getProperty("user.home") + "/latexstudio/editor/resources/latexProperties.txt";
+    
+    public static final String PROPERTIES_DIRECTORY = "latexstudio/editor/resources";
+    
     public static Properties readProperties(String propertyFileName) {
-        Properties properties = new Properties();
-        InputStream is = PropertyService.class.getClassLoader().getResourceAsStream("latexstudio/editor/resources/" + propertyFileName);
+        Properties properties = new Properties(); //start from here
+        //args formerly: "latexstudio/editor/resources/" + propertyFileName
         try {
-            properties.load(is);
-        } catch (IOException ex) {
-            Exceptions.printStackTrace(ex);
+            //formerly InputStream is = PropertyService.class.getClassLoader().getResourceAsStream(PROPERTIES_FILEPATH);
+            FileInputStream fileInputStream = new FileInputStream(PROPERTIES_FILEPATH);
+            properties.load(fileInputStream);
+        } catch (Exception ex) { //may throw either IOException or FileNotFoundException; user shouldn't see either
+            //Exceptions.printStackTrace(ex);
+            return null;
         }
-        
         return properties;
     }
+    
+    public static void writeProperties(Properties toWrite) {
+        try {
+             FileOutputStream fileOutputStream;
+            String homeDirectory = System.getProperty("user.home");
+            File file = new File(homeDirectory + "/" + PROPERTIES_DIRECTORY);
+            file.mkdirs();
+            file = new File(PROPERTIES_FILEPATH);
+            if(file.exists()) {
+                file.delete();
+            }
+            file.createNewFile();
+            fileOutputStream = new FileOutputStream(file);
+            toWrite.store(fileOutputStream, null);
+            fileOutputStream.close();

```

Comment:
streams should be closed in the finally block (since currently it will not get executed in case of exception, which will leave us with opened stream)